src/request_body_processor/xml.cc

```diff
@@ -25,11 +25,122 @@ namespace RequestBodyProcessor {
+    : nodes{},
+    node_depth(0),
+    currpath(""),
+    currval(""),
+    m_transaction(transaction)
+    {}
+
+XMLNodes::~XMLNodes() {};
+
+/*
+* SAX handler for parsing XML into args
+*/
+class MSCSAXHandler {
+    public:
+        void onStartElement(void * ctx, const xmlChar *localname) {
+
+            std::string name = reinterpret_cast<const char*>(localname);
+
+            XMLNodes* xml_data = static_cast<XMLNodes*>(ctx);
+            xml_data->nodes.push_back(std::make_shared<NodeData>());
+            xml_data->node_depth++;
+            // FIXME - later if we want to check the depth of XML tree
+            /* if (max_depth > 0 && max_depth > xml_data->node_depth) {
+                std::cout << "Depth of XML tree reached the given maximum value " << xml_data->node_depth << std::endl;
+                exit(1);
+            } */
+            // if it's not the first (root) item, then append a '.'
+            // note, this can't occur because there is always a pseudo root element: 'xml'
+            if (xml_data->nodes.size() > 1) {
+                xml_data->currpath.append(".");
+                xml_data->nodes[xml_data->nodes.size()-1]->has_child = true;
+            }
+            xml_data->currpath.append(name);
+        }
+
+        void onEndElement(void * ctx, const xmlChar *localname) {
+            std::string name = reinterpret_cast<const char*>(localname);
+            XMLNodes* xml_data = static_cast<XMLNodes*>(ctx);
+            const std::shared_ptr<NodeData>& nd = xml_data->nodes[xml_data->nodes.size()-1];
+            if (nd->has_child == true) {
+                // check the return value
+                // if it false, then stop parsing
+                // this means the number of arguments reached the limit
+                if (xml_data->m_transaction->addArgument("XML", xml_data->currpath, xml_data->currval, 0) == false) {
+                    xmlStopParser(xml_data->parsing_ctx_arg);
+                }
+            }
+            if (xml_data->currpath.length() > 0) {
+                // set an offset to store this is the first item or not -> remove the '.' or not
+                int offset = (xml_data->nodes.size() > 1) ? 1 : 0;
+                xml_data->currpath.erase(xml_data->currpath.length() - (name.length()+offset));
+            }
+            xml_data->nodes.pop_back();
+            xml_data->node_depth--;
+        }
+
+        void onCharacters(void *ctx, const xmlChar *ch, int len) {
+            XMLNodes* xml_data = static_cast<XMLNodes*>(ctx);
+            std::string content(reinterpret_cast<const char *>(ch), len);
+
+            xml_data->currval = content;
+        }
+};
+
+extern "C" {
+    void MSC_startElement(void *userData,
+        const xmlChar *name,
+        const xmlChar *prefix,
+        const xmlChar *URI,
+        int nb_namespaces,
+        const xmlChar **namespaces,
+        int nb_attributes,
+        int nb_defaulted,
+        const xmlChar **attributes) {
```

Comment:
Same here on this declaration. Is this the new standard?

Reply:
Sorry, could you show a "reference"? What is the "standard"? What is the "old" standard?

Reply:
The signature of the function is just in the same line, normally.

Reply:
As I had pointed out in the other PR, there are already several different ways of formatting long function signatures in the code base. This is one of them.